server/src/main/java/org/apache/iotdb/db/engine/snapshot/SnapshotLoader.java

```diff
@@ -361,24 +429,44 @@ public List<File> getSnapshotFileInfo() throws IOException {
   private List<File> getSnapshotFileWithLog(File logFile) throws IOException {
     SnapshotLogAnalyzer analyzer = new SnapshotLogAnalyzer(logFile);
     try {
-
+      String snapshotId = analyzer.getSnapshotId();
+      String[] dataDirs = IoTDBDescriptor.getInstance().getConfig().getDataDirs();
       List<File> fileList = new LinkedList<>();
-
-      while (analyzer.hasNext()) {
-        fileList.add(new File(analyzer.getNextPairs().right));
+      for (String dataDir : dataDirs) {
+        String snapshotDir =
+            dataDir
+                + File.separator
+                + IoTDBConstant.SNAPSHOT_FOLDER_NAME
+                + File.separator
+                + snapshotId;
+        fileList.addAll(searchDataFilesRecursively(snapshotDir));
       }
-
       return fileList;
     } finally {
       analyzer.close();
     }
   }
 
-  private List<File> getSnapshotFileWithoutLog() {
-    return new LinkedList<>(
-        Arrays.asList(
-            Objects.requireNonNull(
-                new File(snapshotPath)
-                    .listFiles((dir, name) -> SnapshotFileSet.isDataFile(new File(dir, name))))));
+  /**
+   * Search all data files in one directory recursively.
+   *
+   * @return
+   */
+  private List<File> searchDataFilesRecursively(String dir) {
```

Comment:
We can use [Files.walkFileTree](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#walkFileTree(java.nio.file.Path,%20java.nio.file.FileVisitor)) here.